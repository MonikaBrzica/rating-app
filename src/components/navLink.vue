<template>
  <router-link tag="li"
               :to="this.data.link"
               active-class="active">
    <div class="item" v-on:click="itemClicked(data.id)">
      <img :src="require('../assets/images/'+ this.data.src)"
           alt="Emoticon">
      <p>{{this.data.text}}</p>
    </div>
  </router-link>
</template>
<script>
export default {
  name: 'NavLink',
  props: {
    data: Array
  },
  data: function () {
    return {
    }
  },
  methods: {
    itemClicked: function (id) {
      if (id === 3) {
        this.$emit('show')
      }
    }
  }
}
</script>
<style lang="scss" scoped>
li.router-link-active,
li.router-link-exact-active {
  background-color: $bg-light;
  opacity: 1;
}
a{
  text-decoration: none;
}
li{
  list-style: none;
  opacity: 0.5;
  .item {
    width: 100px;
    height: 72px;
    text-align: center;
    padding-top: 10px;
    img {
      margin-top: 10px;
    }
    p {
      font-family: 'Roboto-Medium',sans-serif;
      text-align: center;
      font-size: 0.75rem;
      word-spacing: 1rem;
      color: $white-850;
      margin: 0;
    }
  }
}

@media only screen and (max-width: 768px) {
  li {
    width: 64px;
    height: 60px;
    display: inline;
    .item {
      width: 64px;
      height: 60px;
      padding-top: 0;
      p {
        font-family: 'Roboto-Medium',sans-serif;
        font-size: 10px;
        word-spacing: 10px;
      }
    }
  }
}

</style>
