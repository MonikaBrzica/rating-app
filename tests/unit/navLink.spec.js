import { shallowMount, createLocalVue } from '@vue/test-utils'
import navLink from '../../src/components/navLink'
import { RouterLinkStub } from '@vue/test-utils'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  actions: {
  },
  state: {
  }
})

describe('navLink', () => {
  const wrapper = shallowMount(navLink, {
    store,
    localVue,
    propsData: {
      data: {
        id: 1,
        text: 'Today',
        src: 'view-dashboard.svg',
        link: '/today'
      }
    },
    stubs: {
      RouterLink: RouterLinkStub
  }
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
