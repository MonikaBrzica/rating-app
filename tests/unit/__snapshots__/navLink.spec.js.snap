// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navLink renders correctly 1`] = `
<li>
  <div
    class="item"
  >
    <img
      alt="Emoticon"
      src=""
    />
     
    <p>
      Today
    </p>
  </div>
</li>
`;
